### ISSUE TITLE
When presence is disabled for a server show something other than 'Offline'

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#4398]
This likely requires some backend code, but it does alleviate the increasingly common question of "Why am I offline?". Instead of showing people as offline, show them as 'Unknown' or something less concrete to indicate the uncertainty.

**Comment by MTRNord** [comment]
Maybe just not show it if it is disabled?

**Comment by t3chguy** [comment]
The issue is that the client doesn't know that its disabled, its a part of synapse disabled and it still functions over federation (incoming) so would not be quite as simple @MTRNord

**Comment by turt2live** [comment]
Related: #2667

**Comment by lampholder** [comment]
We keep seeing bugs/rageshakes/etc. about this - is there anything we could to so that the whole feature feels less broken for casual users?

**Comment by turt2live** [comment]
I can't see the rageshakes, but idling in the Riot rooms seems to suggest most of the reports revolve around people seeing themselves as offline (and then everyone else). A hopefully-quick improvement could be to always show the current user as online, regardless of the homeserver's suggestion.

For the second part, the homeserver exposing a flag for whether or not it is tracking presence would probably be needed for a proper solution, although you might be able to get away with just showing everyone else as online, effectively removing presence in Riot (probably not a good option).

**Comment by DJCrashdummy** [comment]
> [...] the homeserver exposing a flag for whether or not it is tracking presence would probably be needed for a proper solution [...]

it seems to me to be the only proper solution. :+1:

**Comment by hex-m** [comment]
We already use `.well-known/matrix/client` to communicate things like e2ee-by-default, preferred Jitsi Meet instance, preferred identity server and preferred integration managers to the clients. This would seem like the logical place to tell clients that a server does not handle presence.

**Comment by t3chguy** [comment]
Yup. But would require an MSC given that atm presence isn't even an optional feature of the protocol.

**Comment by aaronraimist** [comment]
https://github.com/matrix-org/matrix-doc/pull/1862 is the MSC (uses `/_matrix/client/r0/capabilities` instead of well-known but very similar idea)
